core/src/main/scala/akka/kafka/internal/SingleSourceLogic.scala

```diff
@@ -46,7 +46,7 @@ import scala.concurrent.{Future, Promise}
       log.log(partitionLogLevel, "Revoked partitions: {}. All partitions: {}", revokedTps, tps)
     }
 
-    def rebalanceListener: KafkaConsumerActor.ListenerCallbacks =
```

Comment:
I see it was earlier `def` as rebalanceListener would not need to be evaluated in the case of a `ManualSubscription`. Even in this case, wouldn't `lazy val` be better suited?

Reply:
It seems like `rebalanceListener`, `partitionAssignedCB` and `partitionRevokedCB` may or may not be called and `def` on all would be preferred.